src/software/geom/polygon.cpp

```diff
@@ -76,6 +76,52 @@ Polygon Polygon::expand(double expansion_amount) const
     return Polygon(expanded_points);
 }
 
+Polygon Polygon::fromSegment(const Segment& segment, const double radius)
+{
+    /*   The Polygon is constructed as follows:
+     *
+     *        start_l                start_r
+     *           +----------+----------+
+     *           |          |          |
+     *           |          | radius   |
+     *           |          |          |
+     *           +   start  X          +
+     *           |          |          |
+     *           |          |          |
+     *           |          |          |
+     *           |       segment       |
+     *           |          |          |
+     *           |          |          |
+     *           |          |   radius |
+     *           +   end    X----------+
+     *           |                     |
+     *           |                     |
+     *           |                     |
+     *           +----------+----------+
+     *         end_l                 end_r
+     */
+
+    Vector start_to_end = segment.getEnd().toVector() - segment.getStart().toVector();
+    Vector end_to_start = -start_to_end;
+
+    Point end_l = segment.getEnd() + (start_to_end.normalize(radius) +
+                                      start_to_end.perpendicular().normalize(radius));
+    Point end_r = segment.getEnd() + (start_to_end.normalize(radius) -
+                                      start_to_end.perpendicular().normalize(radius));
+
+    Point start_l = segment.getStart() + (end_to_start.normalize(radius) +
+                                          end_to_start.perpendicular().normalize(radius));
+    Point start_r = segment.getStart() + (end_to_start.normalize(radius) -
+                                          end_to_start.perpendicular().normalize(radius));
+
+    return Polygon({
+        end_l,
+        end_r,
+        start_l,
+        start_r,
```

Comment:
I want to keep them separate but wasn't sure on the ball_shot_threshold, because it's used to check if the ball is being passed or shot. Maybe ball_not_kicked.

I missed that when porting the code, thanks for the catch.